shared/types/form/form.ts

```diff
@@ -180,7 +179,7 @@ export type PublicStorageFormDto = Merge<
     StorageFormDto,
     // Arrays like typeof list have numeric index signatures, so their number key
     // yields the union of all numerically-indexed properties.
-    (typeof STORAGE_PUBLIC_FORM_FIELDS)[number]
+    typeof STORAGE_PUBLIC_FORM_FIELDS[number]
```

Comment:
I can't really tell if this change is intended. From a strict sense, `(typeof xxx)[number]` -> `string` (to be precise, a character from the typeof repr). While `typeof xxx[number]` -> `xxx[indexed]` if `xxx` is indexable.

@hui-jie-lim requesting your confirmation please :). Thanks.

Reply:
did you tag the wrong person LOL but okay I updated it cos `backend-lint` was complaining: https://github.com/opengovsg/FormSG/actions/runs/5652828911/job/15313165263

Reply:
I think that's the author of the PR that made the change for those `typeof` parts.

Reply:
Ahh yes i see it now, my bad oops. How did it get past the backend-lint 🤔

Reply:
Hello, my apologies but these changes are not intended. I think it might have been due to changes made to `FormSG/package.json` where we added shared/ to the lint scripts. 

cc: @KenLSM 
[Ref PR:](https://github.com/opengovsg/FormSG/pull/6534/files)

Reply:
My bad too. Somehow i missed this during the merge 🤔